### PR TITLE
[examples/flax] use Repository API for push_to_hub

### PR DESCRIPTION
**Pull request by patil-suraj**
# What does this PR do?

This PR updates all flax examples scripts to use the `Repository` API for `push_to_hub` and removes all instructions to manually create/clone a repo in the READMEs.

### CODE REVIEW

File: examples/flax/language-modeling/README.md [hunk 1]
```diff
@@ -33,32 +33,10 @@ in Norwegian on a single TPUv3-8 pod.
...
-```
-
-Next we clone the model repository to add the tokenizer and model files.
-
-```
-git clone https://huggingface.co/<your-username>/norwegian-roberta-base
-```
-
-To setup all relevant files for training, let's go into the cloned model directory.
+To setup all relevant files for training, .
```
Comment (sgugger) [inline examples/flax/language-modeling/README.md:36]: This ends rather abruptly.

File: examples/flax/language-modeling/README.md [hunk 2]
```diff
@@ -33,32 +33,10 @@ in Norwegian on a single TPUv3-8 pod.
...
 
 ```bash
-cd norwegian-roberta-base
-```
-
-Next, let's add a symbolic link to the `run_mlm_flax.py`.
```
Comment (patrickvonplaten) [inline examples/flax/language-modeling/README.md:58]: This should stay IMO

File: examples/flax/language-modeling/README.md [hunk 3]
```diff
@@ -282,30 +239,10 @@ The example script uses the 🤗 Datasets library. You can easily customize them
...
-git clone https://huggingface.co/<your-username>/norwegian-t5-base
-```
-
-To setup all relevant files for trairing, let's go into the cloned model directory.
-
-```bash
-cd norwegian-t5-base
-```
-
-Next, let's add a symbolic link to the `run_t5_mlm_flax.py` and `t5_tokenizer_model` scripts.
```
Comment (patrickvonplaten) [inline examples/flax/language-modeling/README.md:304]: Think we should leave this no? Otherwise the user can't execute `run_t5_mlm_flax.py`

**Review by sgugger — APPROVED** [review]
Looking good! Thanks for switching the scripts to that API :-)

File: examples/flax/question-answering/README.md [hunk 4]
```diff
@@ -26,31 +26,6 @@ of the script.
...
-```
-
-Great, we have set up our model repository. During training, we will automatically
-push the training logs and model weights to the repo.
-
-Next, let's add a symbolic link to the `run_qa.py`.
-
-```bash
-export MODEL_DIR="./bert-qa-squad-test"
-ln -s ~/transformers/examples/flax/question-answering/run_qa.py run_qa.py
```
Comment (patrickvonplaten) [inline examples/flax/question-answering/README.md:52]: should leave this IMO

File: examples/flax/summarization/README.md [hunk 5]
```diff
@@ -11,43 +11,12 @@ way which enables simple and efficient model parallelism.
...
-```
-
-Great, we have set up our model repository. During training, we will automatically
-push the training logs and model weights to the repo.
-
-Next, let's add a symbolic link to the `run_summarization_flax.py`.
-
-```bash
-export MODEL_DIR="./bart-base-xsum"
-ln -s ~/transformers/examples/flax/summarization/run_summarization_flax.py run_summarization_flax.py
```
Comment (patrickvonplaten) [inline examples/flax/summarization/README.md:42]: same here

File: examples/flax/text-classification/README.md [hunk 6]
```diff
@@ -21,55 +21,23 @@ limitations under the License.
...
-Great, we have set up our model repository. During training, we will automatically
-push the training logs and model weights to the repo.
-
-Next, let's add a symbolic link to the `run_flax_glue.py`.
+GLUE is made up of a total of 9 different tasks. Here is how to run the script on one of them:
 
 ```bash
 export TASK_NAME=mrpc
-export MODEL_DIR="./bert-glue-mrpc-test"
-ln -s ~/transformers/examples/flax/text-classification/run_flax_glue.py run_flax_glue.py
```
Comment (patrickvonplaten) [inline examples/flax/text-classification/README.md:58]: same here

File: examples/flax/token-classification/README.md [hunk 7]
```diff
@@ -22,31 +22,6 @@ It will either run on a datasets hosted on our hub or with your own text files f
...
-```
-
-Great, we have set up our model repository. During training, we will automatically
-push the training logs and model weights to the repo.
-
-Next, let's add a symbolic link to the `run_flax_ner.py`.
-
-```bash
-export MODEL_DIR="./bert-ner-conll2003-test"
-ln -s ~/transformers/examples/flax/token-classification/run_flax_ner.py run_flax_ner.py
```
Comment (patrickvonplaten) [inline examples/flax/token-classification/README.md:48]: +1

File: examples/flax/vision/README.md [hunk 8]
```diff
@@ -25,37 +25,6 @@ way which enables simple and efficient model parallelism.
...
-```
-
-Great, we have set up our model repository. During training, we will automatically
-push the training logs and model weights to the repo.
-
-Next, let's add a symbolic link to the `run_image_classification_flax.py`.
-
-```bash
-export MODEL_DIR="./vit-base-patch16-imagenette
-ln -s ~/transformers/examples/flax/summarization/run_image_classification_flax.py run_image_classification_flax.py
```
Comment (patrickvonplaten) [inline examples/flax/vision/README.md:56]: +1

**Review by patrickvonplaten — APPROVED** [review]
Thanks a lot for taking care of this - can we leave the symbolic links though so that people can continue following the steps like a recipe?

**Comment by patil-suraj** [comment]
> Thanks a lot for taking care of this - can we leave the symbolic links though so that people can continue following the steps like a recipe?

@patrickvonplaten the issue with this is that, we need to pass "./" or "." as the `output_dir` which does not work with `Repository`. 

What these changes do is that they create a directory for model/tokenize/config in `/tmp` and uses that path to save the `tokenizer` and `config` and use that as the `output_dir`.  So the users should still be able to follow the steps only difference is the script will nbe run from its own directory rather than creating a symlink.

**Review by LysandreJik — COMMENTED** [review]
This looks good to me, nice usage of the `Repository` API. Something that irks me is that now all trainings happen in `/tmp`, which means that:
1. the trainings will disappear on reboot
2. users may run out of disk space very quickly as `/tmp` is usually a `tmpfs` partition which usually has a size of about half of the available RAM.

Also `Repository(".", clone_from="xxx")` should work!

**Comment by sgugger** [comment]
Maybe the `Path(training_args.output_dir).name` should be update to `Path(training_args.output_dir).absolute().name` in that case?

**Comment by patil-suraj** [comment]
Aah, yeah! Thanks Sylvain :)

**Pull request merged**